site/app/models/Notification.php

```diff
@@ -116,6 +116,9 @@ public static function getUrl($core, $metadata_json) {
         if(is_null($metadata)) {
             return null;
         }
+        if (key_exists('url', $metadata)) {
```

Comment:
Notification-related changes are not complete in this PR. Should wait for #4162 to merge.

Reply:
Mark this as WIP to wait for a notification PR.

Reply:
I quickly tested it and did a glance at the code and looks good for the most part i think, i'll do a more through review once 4162 goes in